code/modules/space_management/level_traits.dm

```diff
@@ -61,7 +61,7 @@ GLOBAL_LIST_INIT(default_map_traits, list(CC_TRANSITION_CONFIG))
 
 /proc/level_name_to_num(name)
 	var/datum/space_level/S = GLOB.space_manager.get_zlev_by_name(name)
-	return S.zpos
+	return S?.zpos
```

Comment:
I don't want this to squelch all references to invalid levels. There's 3 things which use this in reference to the mining level, the first two being in [the mapping controller|https://github.com/ParadiseSS13/Paradise/blob/master/code/controllers/subsystem/mapping.dm#L64-L68], the third being in [the shuttle navigation computer|https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/shuttle/navigation_computer.dm#L31]. Could you instead:
1. Change this to specifically return null for the mining level if lavaland is disabled
2. Add a check to the mapping controller to skip the lavaland special mapgen
3. Modify the shuttle docker computer to check the port's z number being non-null before adding it to the ports list

Reply:
I think everything you said makes perfect sense, except for the first point. I believe it shouldn't be `level_name_to_num`'s responsibility to check for something specific like lavaland being disabled. Either callers need to avoid calling this proc in that case, or they need to consider that it could return null. Do you see where I'm coming from?

Reply:
That was mostly from an "avoiding fan-out" perspective - a change should be accounted for in as few places as possible, without messing with something too deeply entwined for what you're manipulating
But tbh keeping that change outside of the space management subsystem should keep it disentangled from the rest of the codebase so that's an acceptable judgement on your part imo

Reply:
Oh this should still not silently fail if given a bad z level